### DISCUSSION TITLE
Exporting to Parquet is taking large memory and time.

### DISCUSSION THREAD

**Discussion by akash-goel** [duckdb/duckdb#6019] [General]
Hi Team ,

when i am trying to export the data of Duck db table ( Table having 64 million records with 500 columns ) to Parquet using the below code than it is taking memory ( more than 25 Gb) and time and system is hanging , How can we convert Duck db table to parquet with better performance ?

con.execute("COPY (SELECT * FROM test_df) TO 'result-snappy.parquet' (FORMAT 'parquet')")

**Comment by Mytherin** [comment]
Could you provide some more context? What is `test_df`? Is it a table, a Pandas DataFrame, etc? What are the data types?

**Reply by akash-goel** [reply]
Hi Mytherin ,

we have 64 million rows and 500 columns . DataType is Varchar for all the fields . I am using the 32 GB RAM system to convert the duck db table into parquet file

we have created the database test.duckdb and created the table test_df in the database , Below code for your reference
  ```
con = duckdb.connect(database='test.duckdb', read_only=False)
con.execute("CREATE OR REPLACE TABLE test_df as SELECT * FROM 'Parquet/df_1.parquet'")
```
After doing some basic read operation / changing the datatype , we want to convert back into Parquet.

Below Code I am using to convert , where test_df is the table in the database which contains 64 million records with 500 columns
#con.execute("COPY test_df TO 'result-snappy.parquet' (FORMAT 'PARQUET', CODEC 'SNAPPY')")

Regards,
Akash
